packages/core/crates/core-network/src/ping.rs

```diff
@@ -253,343 +277,214 @@ fn to_futures_unordered<F>(mut fs: Vec<F>) -> FuturesUnordered<F> {
-    async fn test_ping_peers_multiple_peers_are_pinged_in_parallel() {
-        let peers = vec![
-            PeerId::from_str(PEER_DELAYED_1_MS).ok().unwrap(),
-            PeerId::from_str(PEER_DELAYED_2_MS).ok().unwrap(),
-        ];
-
-        let mut mock = MockPingExternalAPI::new();
-        mock.expect_on_finished_ping()
-            .with(
-                predicate::eq(peers[0].clone()),
-                predicate::function(|x: &crate::types::Result| x.is_ok()),
-            )
-            .return_const(());
-        mock.expect_on_finished_ping()
-            .with(
-                predicate::eq(peers[1].clone()),
-                predicate::function(|x: &crate::types::Result| x.is_ok()),
-            )
-            .return_const(());
-
-        let pinger = Ping::new(simple_ping_config(), Box::new(mock));
-        pinger.ping_peers(peers, &send_ping).await;
-    }
-
-    #[async_std::test]
-    async fn test_ping_peers_should_ping_parallel_only_a_limited_number_of_peers() {
-        let mut config = simple_ping_config();
-        config.max_parallel_pings = 1;
-
-        let ping_delay_first = 10u64;
-        let ping_delay_second = 10u64;
-
-        let peers = vec![
-            PeerId::from_str(PEER_DELAYED_10_MS).ok().unwrap(),
-            PeerId::from_str(PEER_DELAYED_11_MS).ok().unwrap(),
-        ];
-
-        let mut mock = MockPingExternalAPI::new();
-        mock.expect_on_finished_ping()
-            .with(
-                predicate::eq(peers[0].clone()),
-                predicate::function(|x: &crate::types::Result| x.is_ok()),
-            )
-            .return_const(());
-        mock.expect_on_finished_ping()
-            .with(
-                predicate::eq(peers[1].clone()),
-                predicate::function(|x: &crate::types::Result| x.is_ok()),
-            )
-            .return_const(());
-
-        let pinger = Ping::new(config, Box::new(mock));
-
-        let start = current_timestamp();
-        pinger.ping_peers(peers, &send_ping).await;
-        let end = current_timestamp();
-
-        assert_ge!(end - start, ping_delay_first + ping_delay_second)
-    }
-
-    #[async_std::test]
-    async fn test_ping_peer_should_not_ping_all_peers_if_the_max_timeout_is_reached() {
-        let mut config = simple_ping_config();
-        config.timeout = Duration::from_millis(1);
-        config.max_parallel_pings = 1;
-
-        let peers = vec![
-            PeerId::from_str(PEER_DELAYED_10_MS).ok().unwrap(),
-            PeerId::from_str(PEER_DELAYED_11_MS).ok().unwrap(),
-        ];
-
-        let mut mock = MockPingExternalAPI::new();
-        mock.expect_on_finished_ping()
-            .with(
-                predicate::eq(peers[0].clone()),
-                predicate::function(|x: &crate::types::Result| x.is_ok()),
-            )
-            .return_const(());
-
-        let pinger = Ping::new(config, Box::new(mock));
-
-        pinger.ping_peers(peers, &send_ping).await;
-    }
-}
+// #[cfg(test)]
```

Comment:
It'd be nice to have these tests.

Reply:
need to port them as well ;-) not yet done after some minor API changes

Reply:
done